content/1_General/Cpp_Command.mdx

```diff
@@ -159,29 +159,32 @@ If you want to code in (neo)vim, you can install WSL and code through WSL bash.
 
     probably suffices.
 
-4.  You should be able to compile with `g++-#`, where # is the version number (ex. 9). Running the following command
+4.  You should be able to compile with `g++-#`, where # is the version number (e.g., 10). Running the following command
 
     ```
-    g++-9 --version
+    g++-10 --version
     ```
 
     should display something like this:
 
     ```
-    g++-9 (Homebrew GCC 9.2.0_2) 9.2.0
-    Copyright (C) 2019 Free Software Foundation, Inc.
+    g++-10 (Homebrew GCC 10.2.0_2) 10.2.0
+    Copyright (C) 2020 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
     ```
 
-5.  If you want to be able to compile with just `g++`, create a symbolic link as mentioned [here](https://stackoverflow.com/questions/28970935/osx-replace-gcc-version-4-2-1-with-4-9-installed-via-homebrew/28982564#28982564).
+5.  If you want to be able to compile with just `g++`, write a shell alias! Put the following lines into your shell's rc file (`~/.bashrc` if you use `bash`, and `~/.zshrc` if you use `zsh`).
 
-    ```
-    cd /usr/local/bin
-    ln -fs g++-9 g++
+    ```bash
+    alias g++=g++-10
     ```
 
-    `g++ --version` should now output the same thing as `g++-9 --version`.
+    Once you do so, `g++ --version` should now output the same thing as `g++-10 --version`.
+
+    Note: avoid overriding the system `g++` with symlinking or hard-linking as
```

Comment:
>  Note: avoid overriding the system `g++` with symlinking or hard-linking as
    that will almost surely cause problems. Don't worry if you don't know what
    those terms mean.

what problems?

Reply:
System build processes which expect clang.

Reply:
Could you give an example?

(oops why don't they just use `clang++` then)

Reply:
It's a best practice to not override system packages in general, not limited to just `g++`.

Reply:
still waiting on a concrete example

why not override `g++` with the *real* `g++`? :/

Reply:
For the same reason Apple chose to link `g++` to `clang++`—probably not a very good reason, but we will never know. :/ Also I just realized it was `/usr/local/bin` versus `/usr/bin`, but the issue of `PATH` resolution still exists. As such, it will always be safer to only activate this in _interactive_ shells, which is what aliasing in the rc file does. In addition, the creation of such a link in `/usr/local/bin` is now not managed by Homebrew, which may lead to a dangling symlink if the target executable is (re)moved.

As for a concrete example, `g++` is not really seen as much in the wild anymore. So I guess you could say this is overblown, but the fact is it's not best practice for multiple reasons.

Reply:
tbh I don't completely understand all of this but seems alright :)